test/functional/lib/redirect-handler/blockOrObserve.test.ts

```diff
@@ -0,0 +1,100 @@
+import { expect } from 'chai'
+import { before, describe, it } from 'mocha'
+import browserMock from 'sinon-chrome'
+import browser from 'webextension-polyfill'
+import sinon from 'sinon'
+
+import { optionDefaults } from '../../../../add-on/src/lib/options.js'
+import { addRuleToDynamicRuleSetGenerator, isLocalHost } from '../../../../add-on/src/lib/redirect-handler/blockOrObserve'
+import { initState } from '../../../../add-on/src/lib/state.js'
+import DeclarativeNetRequestMock from './declarativeNetRequest.mock.js'
+
+const dynamicRulesConditions = (regexFilter) => ({
+  regexFilter,
+  excludedInitiatorDomains: [],
+  resourceTypes: [
+    'csp_report',
+    'font',
+    'image',
+    'main_frame',
+    'media',
+    'object',
+    'other',
+    'ping',
+    'script',
+    'stylesheet',
+    'sub_frame',
+    'webbundle',
+    'websocket',
+    'webtransport',
+    'xmlhttprequest'
+  ]
+})
+
+describe('lib/redirect-handler/blockOrObserve', () => {
+  before(function () {
+    browserMock.runtime.id = 'testid'
+  })
+
+  describe('isLocalHost', () => {
+    it('should return true for localhost', () => {
+      expect(isLocalHost('http://localhost:8080')).to.be.true
```

Comment:
we should add a test to ensure regex termination is working properly

```suggestion
      expect(isLocalHost('http://localhost')).to.be.true
      expect(isLocalHost('http://localhost:8080')).to.be.true
```